src/components/drawer/drawer-mixin.js

```diff
@@ -0,0 +1,509 @@
+import '@brightspace-ui/core/components/focus-trap/focus-trap.js';
+import '@brightspace-ui/core/helpers/viewport-size.js';
+import { allowBodyScroll, preventBodyScroll } from '@brightspace-ui/core/components/backdrop/backdrop.js';
+import { clearDismissible, setDismissible } from '@brightspace-ui/core/helpers/dismissible.js';
+import { findComposedAncestor, isComposedAncestor } from '@brightspace-ui/core/helpers/dom.js';
+import { forceFocusVisible, getComposedActiveElement, getNextFocusable, tryApplyFocus } from '@brightspace-ui/core/helpers/focus.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { styleMap } from 'lit/directives/style-map.js';
+
+window.D2L = window.D2L || {};
+window.D2L.DrawerMixin = window.D2L.DrawerMixin || {};
+
+const abortAction = 'abort';
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+const presetSizes = { 'xs': 150, 'sm': 200, 'md': 300, 'lg': 400, 'xl': 500 };
+
+/**
+ * @fires d2l-drawer-close
+ * @fires d2l-drawer-open
+ */
+export const DrawerMixin = superclass => class extends RtlMixin(superclass) {
+
+	static get properties() {
+		return {
+			/**
+             * Aria label for drawer component
+             */
+			label: { type: String },
+			/**
+             * Whether or not the drawer is open (true if drawer is open)
+             */
+			open: { type: Boolean, reflect: true },
+			/**
+             * Drawer body positiong (top, right, bottom, left)
+             */
+			position: { type: String, reflect: true },
+			_closeAborted: { type: Boolean }, // Still working on how to animate - may not need
+			_nestedShowing: { type: Boolean },
+			_overflowBottom: { type: Boolean },
+			_overflowTop: { type: Boolean },
+			_parentDrawer: { type: Object },
+			_scroll: { type: Boolean },
+			_state: { type: String, reflect: true },
+		};
+	}
+
+	constructor() {
+		super();
+		this.label = 'Drawer';
+		this.open = false;
```

Comment:
Can this be called `opened` just so that it matches dialog's API?

Also, instead of having `show()` and `hide()` as public methods, can setting the attribute/property be the thing that triggers showing and hiding? The [motivation for that approach is here](https://daylight.d2l.dev/developing/best-practices/methods/#component-architecture).

The component is already working that way, so the change would be to make `_show()` and `_hide()` internal I think and update the docs to tell people to just get the element and call `elem.opened = true`.